src/torrent_info.cpp

```diff
@@ -411,7 +411,8 @@ namespace {
 				}
 			}
 		}
-		else
+
+		if (symlink_path.empty())
```

Comment:
even though the `symlink` flag is set, the symlink itself may be invalid, in which case we need to treat it as a normal file, that needs a "pieces root" entry